### PR TITLE
Support for kubernetes 1.22, with deprecated support for networkingv1beta1

### PR DESCRIPTION
**Pull request by chauhanshubham**
This commit deprecates networkingv1beta1 IngressClass/Ingress in favor
of networkingv1 APIs to claim support for kubernetes 1.22. The networkingv1beta1
APIs have been deprecaed in kubernetes 1.22.
Resolves #AV-122515

### CODE REVIEW

File: buildsettings.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
         "minVersion": "20.1.3"
     },
     "kubernetes": {
-        "maxVersion": "1.21",
-        "minVersion": "1.16"
+        "maxVersion": "1.22",
+        "minVersion": "1.19"
```
Comment (sudswasavi) [inline buildsettings.json:9]: We are reducing the support matrix to 3 releases only from an earlier 5 releases.

Comment (chauhanshubham) [reply buildsettings.json:9]: *6 to 4 releases yea. We are essentially deprecating 1.16,1.17,1.18 and adding support for 1.21. I think we were fine with this though. cc: @apalsule

File: internal/lib/lib.go [hunk 2]
```diff
@@ -1168,7 +1168,7 @@ func ValidateIngressForClass(key string, ingress *networkingv1beta1.Ingress) boo
 	return true
 }
 
-func filterIngressOnClassAnnotation(key string, ingress *networkingv1beta1.Ingress) bool {
+func filterIngressOnClassAnnotation(key string, ingress *networkingv1.Ingress) bool {
 	// If Avi is not the default ingress, then filter on ingress class.
 	if !GetDefaultIngController() {
 		annotations := ingress.GetAnnotations()
```
Comment (akshayhavile) [inline internal/lib/lib.go:1174]: @chauhanshubham : It seems annotation ""kubernetes.io/ingress.class"" has been deprecated and replaced by "spec.IngressClassName" field https://kubernetes.io/blog/2021/07/14/upcoming-changes-in-kubernetes-1-22/#what-to-do. I think we need to change all occurrences for this.

Comment (chauhanshubham) [reply internal/lib/lib.go]: done

Comment (chauhanshubham) [reply internal/lib/lib.go]: yes! done. thanks.
removed annotation based checks from AKO in the latest commit

**Review by monotosh-avi — DISMISSED** [review]

**Comment by chauhanshubham** [comment]
stuck at `Expected — Waiting for status to be reported`, retrying build

**Comment by chauhanshubham** [comment]
build ako

**Comment by anish-avi** [comment]
build ako

**Comment by chauhanshubham** [comment]
build ako

**Review by monotosh-avi — APPROVED** [review]

**Comment by anish-avi** [comment]
build ako

**Review by sudswasavi — APPROVED** [review]

**Comment by chauhanshubham** [comment]
build ako

**Pull request merged**